src/main/java/learnositysdk/request/Init.java

```diff
@@ -348,63 +349,61 @@ private void setServiceOptions() throws Exception
-        
-
+        // In case the user gave us a securityPacket String, convert to a JSONObject
         this.validateSecurityPacket(securityPacket);
-        
+
         if (secret.isEmpty()) {
             throw new Exception("The `secret` argument must be a valid string");
         }
         this.secret = secret;
     }
 
     /**
-     * Validates the request packet argument
-     * @param requestPacket
-     * @throws Exception
+     * Sets the request packet, updating the stored string version as well
+     *
+     * @param requestPacket The request packet to store
      */
-    private void validateRequestPacket(Object requestPacket) throws Exception
+    private void setRequestPacket(Object requestPacket) throws Exception
     {
-        if (requestPacket instanceof JSONObject) {
-            this.requestPacket = new JSONObject(requestPacket.toString());
-            this.requestString = requestPacket.toString();
-        } else {
-            if (requestPacket instanceof String) {
-                this.requestPacket = new JSONObject((String)requestPacket);
-                this.requestString = (String)requestPacket;
-            } else if (requestPacket instanceof Map) {
-                this.requestPacket = new JSONObject((Map)requestPacket);
-                this.requestString = this.requestPacket.toString();
+        JSONObject convertedPacket = validateRequestPacket(requestPacket);
+
+        // Prevent updating request packet in situations where we must not do so
+        if (!this.canAlterRequestData) {
+            // Only properly support using inputs provided as String
+            if (this.requestPacket == null) {
+                if (requestPacket instanceof String) {
+                    this.requestString = (String) requestPacket;
+                    this.requestPacket = convertedPacket;
+                    return;
+                }
+                // If any other format is provided, we cannot make the any
+                // guarantees anyway - fall through.
             } else {
-                // Try to make a JSONObject out of a hopefully valid java bean
-                this.requestPacket = new JSONObject(requestPacket);
-                this.requestString = this.requestPacket.toString();
+                throw new Exception("Cannot update request packet for this request");
             }
         }
 
-        // If requestPacket is not a string, then requestString is generated using
-        // JSONObject.toString that causes escaping issues.
-        if (!(requestPacket instanceof String)) {
-            // JSONObject.toString escapes forward slashes. Undo that, in order to avoid changes to
-            // the string
-            this.requestString = this.requestString.replace("\\/", "/");
-
-            // unescape any escape sequences created by JSONObject.toString
-            this.requestString = StringEscapeUtils.unescapeJava(this.requestString);
```

Comment:
The latest version of the JSON library seems to have resolved issues with characters getting re-escaped unexpectedly. If anyone is able to suggest any extra test cases to confirm there isn't an issue introduced by this change, that'd be super helpful!